vsintegration/src/FSharp.ProjectSystem.Base/Project/ProjectFactory.cs

```diff
@@ -112,6 +112,63 @@ protected override void CreateProject(string fileName, string location, string n
                 }
             }
 
+            // Solution properties
+            IVsSolution solution = this.Site.GetService(typeof(SVsSolution)) as IVsSolution;
+            Debug.Assert(solution != null, "Could not retrieve the solution service from the global service provider");
+
+            // We do not want to throw. If we cannot set the solution related constants we set them to empty string.
+            string solutionDirectory, solutionFile, userOptionsFile;
+            solution.GetSolutionInfo(out solutionDirectory, out solutionFile, out userOptionsFile);
+            if (solutionDirectory == null)
+            {
+                solutionDirectory = String.Empty;
+            }
+            if (solutionFile == null)
+            {
+                solutionFile = String.Empty;
+            }
+            string solutionFileName = (solutionFile.Length == 0) ? String.Empty : Path.GetFileName(solutionFile);
```

Comment:
Can Path.GetFileName or Path.GetFileNameWithoutExtension fail?

Reply:
https://msdn.microsoft.com/en-us/library/system.io.path.getfilenamewithoutextension%28v=vs.110%29.aspx



ArgumentException if there are invalid characters in the name.  However, the solution load could hardly have got this far through.